modules/nf-core/bedtools/map/main.nf

```diff
@@ -0,0 +1,59 @@
+process BEDTOOLS_MAP {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "bioconda::bedtools=2.30.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/bedtools:2.30.0--hc088bd4_0' :
+        'quay.io/biocontainers/bedtools:2.30.0--hc088bd4_0' }"
+
+    input:
+    tuple val(meta), path(intervals1), path(intervals2)
+    tuple val(meta2), path(chrom_sizes)
+
+    output:
+    tuple val(meta), path("*.${extension}"), emit: map
+    path  "versions.yml"                   , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    //Extension of the output file. It is set by the user via "ext.suffix" in the config. Corresponds to the file format which depends on arguments (e. g., ".bed", ".bam", ".txt", etc.).
+    extension = task.ext.suffix ?: "bed"
```

Comment:
You can derive the output extension from the arguments and the input values :) (e.g. when a BAM is given, the output will be BAM unless you specified `-bed`) It would be easier if the module can derive the extension itself from the input and the given arguments instead of having to rely on the user to give them :). If you are stuck on this feel free to ping me